### DISCUSSION TITLE
How can we delete a record in Postgresql based on raw WHERE part

### DISCUSSION THREAD

**Discussion by 1Map** [prisma/prisma#18603] [Q&A] [answered]
Example:

```js
    prisma.RefreshToken.delete({
      where: customWhere(
        `EXTRACT(EPOCH FROM expiry_date AT time zone 'utc') < EXTRACT(EPOCH FROM NOW() AT time zone 'utc')`,
      ),
    })
```

I have tried:

```js
    await prisma.RefreshToken.delete({
      where: {
        expiryDate: {
          lt: new Date()
        }
      }
    });
```

But it throws an error:

```js
Invalid `prisma.refreshToken.delete()` invocation:

{
where: {
    expiryDate: {
    ~~~~~~~~~~
      lt: new Date('2023-04-02T17:37:15.700Z')
    }
  }
}

Unknown arg `expiryDate` in where.expiryDate for type RefreshTokenWhereUniqueInput. Available args:

type RefreshTokenWhereUniqueInput {
  id?: BigInt
  token?: String
}

```

**Comment by ludralph** [comment]
Hi @1Map 👋 

Can you try enabling the preview feature [extendedWhereUnique](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#filter-on-non-unique-fields-with-userwhereuniqueinput) and retry this?

```js
  await prisma.RefreshToken.delete({
      where: {
        expiryDate: {
          lt: new Date()
        }
      }
    });
```

**Reply by ludralph** [reply]
Hi @1Map 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by 1Map** [comment, marked as answer]
@ludralph 

The problem was not what you suggested, but rather:

Cannot delete multiple records with **delete** if you do not use the primary key in your where clause.
This was fixed by using the **deleteMany** as follow:

```
    let timeNow = new Date()
    timeNow.setSeconds(timeNow.getSeconds())

    await prisma.RefreshToken.deleteMany({
      where: {
        expiryDate: {
          lt: timeNow
        }
      }
    });
```
